### ISSUE TITLE
Marathi translation

### ISSUE THREAD

**Issue by NiranjanTambe** [learnlatex/learnlatex.github.io#72]
Hello,

I would like to contribute Marathi translation. I have [forked](https://github.com/NiranjanTambe/learnlatex.github.io) the repository. Kindly let me know how to add a PR.

**Note - The translation is not yet complete. It is under progress.**

**Comment by PhelypeOleinik** [comment]
When you're done translating you can make a PR by going to your fork then clicking the “Pull request” button near the top. In my fork it looks like this:

> ![Screenshot from 2020-07-28 14-26-19](https://user-images.githubusercontent.com/12678598/88700227-c08cee00-d0de-11ea-81d7-e3eb155c2b7d.png)

**Comment by joulev** [comment]
@NiranjanTambe Also if you haven't finished yet, you can mark the PR as "draft", so that it is not merged too early but others are still notified about the progress.

**Comment by davidcarlisle** [comment]
I tried your first example

```
% !TeX xelatex
\documentclass{article}
\usepackage{marathi}

\begin{document}
नमस्कार.
\end{document}
```

and it fails with 

```
! Package fontspec Error: The font "Shobhika" cannot be found.
```

Is there a font that is in a default TeXLive install that works with this package? that would be best as it doesn't require extra resources on the server, but otherwise if there is an open Source font required I could consider installing it.

If you enable githib pages on your fork you should be able to test that your examples work on the server.

**Comment by joulev** [comment]
@davidcarlisle It is in a standard TeX Live installation. https://ctan.org/tex-archive/fonts/shobhika

**Comment by davidcarlisle** [comment]
@joulev  ah fontconfig strikes again :( lualatex finds it I'll configure fontconfig on the server so xelatex finds it by name as well.

**Comment by davidcarlisle** [comment]
![image](https://user-images.githubusercontent.com/1268738/88705775-e5e22200-d107-11ea-82c1-b117bd299b53.png)

**Comment by davidcarlisle** [comment]
@joulev thanks for the correction. fontconfig has now cached the texlive fonts at latexcgi.xyz so the above example now works with xelatex and lualatex using the latex online button at learnlatex.org

**Comment by NiranjanTambe** [comment]
Thanks @joulev !

@davidcarlisle Can you please also cache package [EkType-tanka](https://ctan.org/pkg/ektype-tanka)? It provides five Devanagari fonts, out of which I might need to use some fonts.

**Comment by joulev** [comment]
@NiranjanTambe It should work already as the package is in a standard TeX Live distribution.

**Comment by NiranjanTambe** [comment]
Shobhika also was in a standard TeX Live distribution, but @davidcarlisle had to do something to make it available on this site, right? Don't we need to repeat it for these fonts?

**Comment by joulev** [comment]
@NiranjanTambe AFAIK LaTeXCGI can work for any packages in TL. No special considerations for any packages needed.

**Comment by NiranjanTambe** [comment]
Okay! Thanks

**Comment by Skillmon** [comment]
@NiranjanTambe the problem was that XeLaTeX couldn't find the fonts by name because fontconfig wasn't run on the TeX Live fonts folder, that's what @davidcarlisle fixed to make the initial example work. Now every font contained in TeX Live should work.

**Comment by NiranjanTambe** [comment]
Sounds great :) Thanks for the explanation.

**Comment by davidcarlisle** [comment]
@NiranjanTambe  if you delete the CNAME file and enable github pages (a check box in the settings on your fork) the site should appear at

https://niranjantambe.github.io/learnlatex.github.io/

and you can test the examples work directly.

**Comment by NiranjanTambe** [comment]
@davidcarlisle Thanks.
I removed CNAME and enabled GitHub pages. I can now see the page, but still I can't see `mr` in the language selection tool. What do I need to do for that?

**Comment by davidcarlisle** [comment]
see https://www.learnlatex.org/TRANSLATIONS add `mr` to the list in `_config.yml` should be all you need

**Comment by davidcarlisle** [comment]
meanwhile I can see the buttons at https://niranjantambe.github.io/learnlatex.github.io/mr/lesson-03  but can no longer read them:-)

**Comment by NiranjanTambe** [comment]
> see https://www.learnlatex.org/TRANSLATIONS add mr to the list in _config.yml should be all you need

Does it take some time to adjust? My browser is redirecting me to the homepage when I open https://niranjantambe.github.io/learnlatex.github.io/mr

and yes I have translated the buttons too. I hope it's fine :)

**Comment by NiranjanTambe** [comment]
Oh I have received a warning in my mailbox from GitHub, which says -

> The page build completed successfully, but returned the following warning for the `master` branch:
 The CNAME `www.learnlatex.org` is already taken. Check out https://help.github.com/articles/troubleshooting-custom-domains#cname-errors for more information.

Did I do something wrong in my code? How to rebuild the whole thing without the CNAME file?

**Comment by davidcarlisle** [comment]
looks OK as far as I can see but I can not see your github repository settings, does it have the custom DNS set to learnlatex.org?  if so uncheck that and let it use the default github address, either that or the mail you git was simply delayed from the initial build before you deleted the CNAME.

**Comment by davidcarlisle** [comment]
https://niranjantambe.github.io/learnlatex.github.io/mr/help

works and the link in there to 


https://niranjantambe.github.io/learnlatex.github.io/mr/more-14

works

That is probably enough for you to do the translation and we can fix up any remaining things once it is merged back to the main site but the one thing that isn't working is your main index page

https://niranjantambe.github.io/learnlatex.github.io/mr/

is acting line the english one and redirecting to the (English) top level page.

AH!! I see what is wrong. You followed the instructions in TRANSLATIONS but they are wrong:-)

Delete the file mr/index.html  that is a small stub html file that forces a redirect to the top level so that

https://www.learnlatex.org/en is the same as https://www.learnlatex.org

That is only required for English.

In mr you have both index.html and index.md but the index.md one is not used (as it would make an index,html which already exists) so if you delete mr/index.html then the site should use your index.md file and things will work better hopefully.

Sorry about that.

**Comment by NiranjanTambe** [comment]
I removed the html file and now the index page works. Thanks.

I have one more request. Can you please install the latest version of package [`marathi`](https://ctan.org/pkg/marathi) (v1.3)? I also checked TeX Live repository and the latest version is released there too. You can find it [here](https://tug.org/svn/texlive/trunk/Master/texmf-dist/tex/latex/marathi/). I need it for [this](https://niranjantambe.github.io/learnlatex.github.io/mr/extra-01#काव्य) example.

**Comment by davidcarlisle** [comment]
@NiranjanTambe  I did a tlmgr update a couple of days ago I will do one again now then latexcgi.xyz will have whatever is in texlive 2020.

overleaf is texlive 2019 (until later in the year) so marathi.sty is not there but I just  adapted the javascript to add it to the projects 


If you try


https://davidcarlisle.github.io/ll/mr/help

then all four buttons should work, that is xelatex and lualatex on overleaf and latexcgi.xyz

To get the overleaf buttons to work you need the mr/includes/buttons.js  from

https://github.com/davidcarlisle/ll/blob/master/mr/includes/buttons.js

That extends the overleaf submisision to add marathi.sty (it may not be the latest version, you can check:-)

speaking of versions it would be helpful if the package did

```
\ProvidesPackage{marathi}[2020-07-29  v1.5 Marathi Support]
```

or similar that way the version of the package used is recorded in the latex log file. You do not have the optional argument currently.


... I just updated texlive at latexcgi.xyz but marathi.sty no longer works in luatatex:

```
! Undefined control sequence.
l.92 \defaultfontfeatures
                       [\rmfamily,\sffamily,\ttfamily]{Script=Devanagari,Ren...
? 
! Emergency stop.
l.92 \defaultfontfeatures
                       [\rmfamily,\sffamily,\ttfamily]{Script=Devanagari,Ren...
End of file on the terminal!
```

I get the same on my local system.

**Comment by joulev** [comment]
@NiranjanTambe Your package is good, however as this is the guide for absolute newbies, we want to make sure all instructions are best practices and have been verified and frequently used by a good number of users.

I am afraid your package is a bit too new to be considered the best practice by all Indian TeX users. It is also more unstable than it should be (see @davidcarlisle's message above).

Therefore I suggest a mr/language-01 having a section showing the way that most Indian TeX users are using for Marathi in LaTeX. Description about `marathi` can be put to a different section under the same language-01 or a separate language-02 if you want to.

**Comment by joulev** [comment]
> I need it for [this](https://niranjantambe.github.io/learnlatex.github.io/mr/extra-01#%E0%A4%95%E0%A4%BE%E0%A4%B5%E0%A5%8D%E0%A4%AF) example.

@NiranjanTambe I think, in all pages other than the language-specific ones, you should keep the code part unmodified and only translate the comments. Not sure what the main site designers think though. @josephwright @davidcarlisle

**Comment by josephwright** [comment]
Indeed: as far as possible, the 'main thread' lessons should have the same content.

**Comment by davidcarlisle** [comment]
@joulev in the translations instructions we (currently) give the option of translating or not translating the examples. I think it depends on the community and how familiar a new latex user is likely to be with English. It does complicate our lives a bit but I think it's good to have the possibility. I am not sure for example which way the Japanese translation will go. We could ask @wtsnjp  what his intentions are there....

**Comment by NiranjanTambe** [comment]
@davidcarlisle 

> That extends the overleaf submisision to add marathi.sty (it may not be the latest version, you can check:-)

Done now! Thank you so much for helping me with this. I was worried whether my overleaf button will work or not.

I'll add the optional argument to the package name in the next release which will solve the lualatex problem too. (Thanks for the [help](https://gitlab.com/niranjanvikastambe/marathi/-/commit/5e4e91fa4e0ace7f9eed86986872743d10dc7e85#note_387897997).)

@joulev 

As @davidcarlisle pointed out in his message translation instructions allow me to have Marathi examples "along with" the English ones. I am doing that only everywhere else. I have only changed the poetry example as it is absolutely unnatural in Marathi to write poems that way. Hence the example is completely un-relatable for us. Therefore I am changing it to a more practical one. I hope that sounds fine.

> I suggest a mr/language-01 having a section showing the way that most Indian TeX users are using for Marathi in LaTeX.

This is a good idea. I would add a lesson about it as an _alternate_ way of typesetting Marathi, but note that there is no standard way! I will add a method based on my knowledge only.

@josephwright 

> Indeed: as far as possible, the 'main thread' lessons should have the same content.

Yes. I would remove the changes that I have made to other examples, but I request you to at least consider the poetry example as an exception. That example is unpractical for not only Marathi users, but for many other non-English TeX-users.

**Comment by davidcarlisle** [comment]
Thank you  for working on learnlatex.org translations.
You may have seen already if watching the repository but just a note
to say that we have extended lesson-06 and more-06 to add descriptions
of defining commands with \newcommand and \NewDocumentCommand

https://github.com/learnlatex/learnlatex.github.io/commit/463d181301d667b10bf23c164f9d5ae15371877e

**Comment by NiranjanTambe** [comment]
Thanks @davidcarlisle,

I am stuck in some work and hence couldn't check the repository very actively. Hence it's good that you informed me about it here. Can you tell please tell me what is the correct way of adding these new changes to my local repository? I don't want to lose my work that has been already done, but also want new changes.

**Comment by davidcarlisle** [comment]
@NiranjanTambe  if you have checked in your local changes (`git status -uno` says you have no unsaved changes) then whatever happens you will not lose work you will be able to restore things from the repository

To resync your fork you can do

```
git fetch upstream
git checkout master
git merge upstream/master
```

the first two lines just pull the files into your local repository they will not affect your working directory at all.
The last line merges the files to your working directory so could in principle have some merge conflicts that need resolving, but as all your changes are in `mr` and no one else has made changes in there I do not think you should have issues.

The github site lets you generate diffs between any two versions if you want to see what has changed, see for example the link in the comment I gave above, but you could generate similar diff for the English versions of any files you have already translated, to check no major changes have been added.

**Comment by joulev** [comment]
@davidcarlisle I use

```
git pull https://github.com/learnlatex/learnlatex.github.io.git master
```

instead. It works, but is it better or worse than the commands you mentioned above?

**Comment by davidcarlisle** [comment]
when you generated the fork at github originally github will have set up an "upstream"  remote and an upstream/master branch being short names for the original learnlatex repository, so (I think) that referencing it explictly you end up with two local copies  of that but to be honest I'm not sure, maybe git notices that they are the same thing so it makes no difference, git is good at noticing such things:-)

**Comment by NiranjanTambe** [comment]
@davidcarlisle 
On my machine I got this error for your first command.

```
git fetch upstream
fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.
```

**Comment by davidcarlisle** [comment]
@NiranjanTambe  yes sorry I may have mislead you, I did check one of my forks but maybe I had previously named the remote, I thought github did it when making the fork. I'll check the docs again and may edit that comment to save misleading any other readers.

**Comment by davidcarlisle** [comment]
@NiranjanTambe  Sorry it seemed I had added this in the forks I had checked  previously.
It seems that github doesn't make the remote on forks automatically, although many of its doc pages assume that there is a named remote, the above assumes that you have

git remote add upstream git@github.com:learnlatex/learnlatex.github.io.git

But it's just making `upstream` essentially an alias for what you did. Sorry for the confusion.

**Comment by NiranjanTambe** [comment]
Thanks for this @davidcarlisle,
but this resulted in one error at the last command. Here is the log -

```
Auto-merging _config.yml
CONFLICT (content): Merge conflict in _config.yml
Auto-merging AUTHORS.md
Automatic merge failed; fix conflicts and then commit the result.
```

**Comment by davidcarlisle** [comment]
@NiranjanTambe  that's probably OK, if you have adjusted the list of languages and list of authors and others have made similar changes in other branches there will be easy to fix merge conflicts there, just look in those files remove the `>>>` sections to make a correct list again and then commit the changed files.

If you complete the translation and want to merge it all back to learnlatex it's quite likely that I get merge conflicts in similar files, but I will just fix up th elist here to expose whatever lists of languages we want public at that time.

**Comment by davidcarlisle** [comment]
@NiranjanTambe  I haven't seen an official announcement yet, but Overleaf is now using Texlive 2020 for new projects and so includes marathi.sty. This means that you can delete the special version of the openinoverleaf function in mr/includes/buttons.js that uploaded a local copy to each project see

https://www.overleaf.com/read/chkkmrymnwsq

which has  use of your package with no special configuration in the project.

**Comment by NiranjanTambe** [comment]
Thanks for informing me about that. I will change the code in my local repository.

**Comment by davidcarlisle** [comment]
Hi @NiranjanTambe  I hope you are keeping well in these difficult times.

As you may have seen, the main site at learnlatex.org has been given a new design and some of the details of the language data have changed. 

If you are still working on the translation, I have updated
https://www.learnlatex.org/TRANSLATIONS
to hopefully give correct instructions for the new layout, but if anything is not clear please ask

**Comment by NiranjanTambe** [comment]
Hi @davidcarlisle 

I am still interested in the translations. I made my old files compatible with the new design yesterday! I need more time for the translations, but till then I need to report some bugs that I found on the site. They are as follows.

(My website is available at - https://niranjantambe.github.io/learnlatex.github.io/mr/)

1. Table of contents on the homepage doesn't show Marathi headings. I don't know from where it is taking the English headings. In my code I have followed the method used for the English lessons as far as I see. Kindly let me know if I have done something wrong there. Also this is not uniform. I can see [this](https://niranjantambe.github.io/learnlatex.github.io/mr/#मराठी-भाषाविशिष्ट-सोयी) correctly in the table of contents, but not [this](https://niranjantambe.github.io/learnlatex.github.io/mr/#1-latex-basics) one.
2. I can't see the navigation icons which are correctly rendered in the other translations of the site. See the following screenshot.

![Screenshot_2021-02-04_09-20-49](https://user-images.githubusercontent.com/54938729/106842599-54f98e00-66ca-11eb-957e-e149576a42cb.png)

3. Marathi translation of the website shall have Devanagari numerals instead of Arabic numerals. e.g. in the screenshot attached above, `प्रकरण 1` should be `प्रकरण १`. If you want list of Devanagari numerals (ranging from 0-9) it is `०,१,२,३,४,५,६,७,८,९`. Kindly help me in this matter. Auto-rendered numerals should be in Devanagari.

Thanks.

**Comment by NiranjanTambe** [comment]
Thanks,

This translation is now merged in the master, so closing this issue. Kindly reopen it if any problems are found in the code.

**Comment by davidcarlisle** [comment]
@NiranjanTambe  I just noticed (while checking something for Japanese) that 

https://www.learnlatex.org/mr/lesson-11

the first paragraph seems to gave been missed and is still in English, it's just a couple of lines. If you could supply the text either as a PR or just add it here and I'll drop it in, thanks.

**Comment by davidcarlisle** [comment]
@NiranjanTambe  in addition to the one paragraph of lesson-11, the whole of more-15 was missed in the orginal translation.


https://www.learnlatex.org/mr/more-15


Also, if possible, it would be nice to get translations for two new files that were added

https://www.learnlatex.org/mr/settings

and 

https://www.learnlatex.org/mr/privacy

along with translations in `translations.yml` of the `Settings` and `Privacy` words that appear in navigational links on every page.

There is of course no rush or deadline, but as your translation is so nearly complete it would be nice to not have the english navigational elements.

Thanks for your work on this so far,

David

**Comment by NiranjanTambe** [comment]
I am extremely sorry for the late response.

I saw your [SE ping](https://tex.stackexchange.com/questions/594232/renewdocumentcommand-t-doesnt-work-with-package-hyperref?noredirect=1#comment1501535_594232), but I was already trying to work on the translations. I am taking help from some friends of mine for some technical words, hence it is taking time. I will need some more time for completing the translations. I thought rather than pinging you again and again I'll just complete the translations and add them, but I am sorry for a long silence from my side. I'll work upon these files and add them soon.

Thanks

**Comment by davidcarlisle** [comment]
Sorry about that, No worry at all about the length of time. I just wondered
if you'd missed the github ping (as you are mostly at gitlab) hence I
nudged you on stackexchange:-)

David
